haas/api.py

```diff
@@ -237,6 +237,10 @@ def node_register(node, **kwargs):
         raise BadArgumentError('%r is not a valid OBM type.' % obm_type)
     cls.validate(kwargs['obm'])
     node_obj = model.Node(label=node, obm=cls(**kwargs['obm']))
+    if 'metadata' in kwargs:
```

Comment:
I think you'll need to add `metadata` as an optional parameter to the schema above.